### PR TITLE
[risk=low][RW-7533] Disallow CT workspace sharing with users who are no longer in CT

### PR DESCRIPTION
**Pull request by jmthibault79**
Missed a step in the search query: check if the user/tier association is ENABLED rather than DISABLED.

Tested locally: could share with non-CT user before, can only share with CT users now.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/db/dao/UserDaoTest.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ public void testGetAdminTableUsers() throws Exception {
     final DbInstitution institution = createInstitution();
     user = userDao.save(user);
     createAffiliation(user, institution);
-    addUserToTier(user, registeredTier, TierAccessStatus.ENABLED);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/db/dao/UserDaoTest.java:174]: unrelated cleanup

File: api/src/test/java/org/pmiops/workbench/db/dao/UserDaoTest.java [hunk 2]
```diff
@@ -415,10 +465,6 @@ public DbUserAccessModule addUserAccessModule(
             .setCompletionTime(completionTime));
   }
 
-  private DbUserAccessTier addUserToTier(DbUser user, DbAccessTier tier) {
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/db/dao/UserDaoTest.java:418]: moved up to group with the other version of this method

**Review by aweng98 — APPROVED** [review]
LGTM

**Review by NehaBroad — APPROVED** [review]
